AdminUI/Controllers/HomeController.cs

```diff
@@ -157,6 +157,50 @@ public IActionResult Timesheet(int ID)
             return View();
         }
 
+        public FileContentResult DownloadCSV(string pName, string cName, string dateFrom, string dateTo, string prime, string id)
+        {
+            var sheets = GetSheets();
+
+            
+            //filter the timesheets 
+            if (!string.IsNullOrEmpty(pName))
+                sheets = sheets.Where(t => t.ProviderName.ToLower().Contains(pName.ToLower()));
+
+            if (!string.IsNullOrEmpty(cName))
+                sheets = sheets.Where(t => t.ClientName.ToLower().Contains(cName.ToLower()));
+
+            if (!string.IsNullOrEmpty(dateFrom))
+                sheets = sheets.Where(t => t.Submitted >= DateTime.Parse(dateFrom));
+
+            if (!string.IsNullOrEmpty(dateTo))
+                sheets = sheets.Where(t => t.Submitted <= DateTime.Parse(dateTo));
+
+            if (!string.IsNullOrEmpty(prime))
+                sheets = sheets.Where(t => t.ClientPrime == prime);
+
+            if (!string.IsNullOrEmpty(id))
+                sheets = sheets.Where(t => t.TimesheetID == int.Parse(id));
+
+            //the following loops through every property in a timesheet, first saving the names of the properties to 
+            //act as a header. Then, it loops through every timesheet, adding every individual property of the timesheet
+            //to the csv, then returning it for download.
+            var properties = typeof(Timesheet).GetProperties();
+            var csv = properties.Aggregate("", (current, f) => current + (f.Name + ','));
+            foreach (var s in sheets)
+            {
+                csv += '\n';
+                foreach (var p in properties)
+                {
+                    if (p.GetValue(s) != null)
+                        csv += "\"" + p.GetValue(s) + "\"";
```

Comment:
A potential issue is if a field happens to have a double quote already in it.

Reply:
I added some quotes and commas to see what would happen and boy it did not like that at all. I'll close this request, fix that, and open a new one

![image](https://user-images.githubusercontent.com/49330987/79150589-e71c5880-7d7d-11ea-851b-add9d3903c52.png)


![image](https://user-images.githubusercontent.com/49330987/79150561-d5d34c00-7d7d-11ea-972e-f7bbce634082.png)

Reply:
Never mind, it was an easier fix than I realized. Replacing double quotes in the string with single quotes fixed it right up

![image](https://user-images.githubusercontent.com/49330987/79150879-5a25cf00-7d7e-11ea-8163-b37491060794.png)

![image](https://user-images.githubusercontent.com/49330987/79150902-6742be00-7d7e-11ea-8075-1859d5123be0.png)